File: Selenium/Activities/Activity17.java
package activities;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {
    public static void main(String[] args) {
        // Create a new instance of the Chrome driver
        WebDriver driver = new ChromeDriver();

        // Open the page
        driver.get("https://training-support.net/webelements/selects");
        // Print the title of the page
        System.out.println("Page title: " + driver.getTitle());

        // Find the dropdown
        WebElement selectElement = driver.findElement(By.cssSelector("select.h-80"));
        // Pass the WebElement to the Select object
        Select multiSelect = new Select(selectElement);

        // Select "HTML" using visible text
        multiSelect.selectByVisibleText("HTML");
        // Select 4th, 5th, and 6th index options
        for (int i = 3; i <= 5; i++) {
            multiSelect.selectByIndex(i);
        }
        // Select "Node" using value attribute
        multiSelect.selectByValue("nodejs");
        // Print the selected options
        List<WebElement> selectedOptions = multiSelect.getAllSelectedOptions();
        System.out.println("Selected options are: ");
        for (WebElement option : selectedOptions) {
            System.out.println(option.getText());
        }

        // Deselect the 5th index option
        multiSelect.deselectByIndex(4);
        // Print the selected options
        selectedOptions = multiSelect.getAllSelectedOptions();
        System.out.println("Selected options are: ");
        for (WebElement option : selectedOptions) {
            System.out.println(option.getText());
        }

        // Close the browser
        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

# Start the Driver
with webdriver.Firefox() as driver:
    # Navigate to the URL
    driver.get("https://training-support.net/webelements/selects")
    # Print the title of the page
    print("Page title is: ", driver.title)

    # Find the dropdown
    selectElement = driver.find_element(By.CSS_SELECTOR, "select.h-80")
    # Pass the WebElement to the Select object
    multiSelect = Select(selectElement)

    # Select "HTML" using visible text
    multiSelect.select_by_visible_text("HTML")
    # Select 4th, 5th, and 6th index options
    for i in range(3, 5):
        multiSelect.select_by_index(i)
    # Select "Node" using value attribute
    multiSelect.select_by_value("nodejs")
    # Print the selected options
    selectedOptions = multiSelect.all_selected_options
    print("Selected options are: ")
    for option in selectedOptions:
        print(option.text)

    # Deselect the 5th index option
    multiSelect.deselect_by_index(4)
    # Print the selected options
    selectedOptions = multiSelect.all_selected_options
    print("Selected options are: ")
    for option in selectedOptions:
        print(option.text)